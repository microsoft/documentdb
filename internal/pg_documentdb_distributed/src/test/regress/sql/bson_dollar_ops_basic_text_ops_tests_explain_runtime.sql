
set search_path to helio_core,helio_api,helio_api_catalog,pg_catalog;
SET citus.next_shard_id TO 1400000;
SET helio_api.next_collection_id TO 1400;
SET helio_api.next_collection_index_id TO 1400;

SELECT helio_api.create_collection('db', 'queryregexopstest');
SELECT helio_api.create_collection('db', 'querytextopstest');

\i sql/bson_dollar_ops_basic_text_ops_tests_explain_core.sql
