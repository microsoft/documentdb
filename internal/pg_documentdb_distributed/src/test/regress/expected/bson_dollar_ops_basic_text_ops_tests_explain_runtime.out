set search_path to helio_core,helio_api,helio_api_catalog,pg_catalog;
SET citus.next_shard_id TO 1400000;
SET helio_api.next_collection_id TO 1400;
SET helio_api.next_collection_index_id TO 1400;
SELECT helio_api.create_collection('db', 'queryregexopstest');
NOTICE:  creating collection
 create_collection 
---------------------------------------------------------------------
 t
(1 row)

SELECT helio_api.create_collection('db', 'querytextopstest');
NOTICE:  creating collection
 create_collection 
---------------------------------------------------------------------
 t
(1 row)

\i sql/bson_dollar_ops_basic_text_ops_tests_explain_core.sql
/* validate explain */
EXPLAIN (COSTS OFF) SELECT object_id, document FROM helio_api.collection('db','queryregexopstest') WHERE document @~ '{ "a.b": "^.+$" }';
                                              QUERY PLAN                                               
---------------------------------------------------------------------
 Custom Scan (Citus Adaptive)
   Task Count: 1
   Tasks Shown: All
   ->  Task
         Node: host=localhost port=58070 dbname=regression
         ->  Bitmap Heap Scan on documents_1400_1400007 collection
               Recheck Cond: (shard_key_value = '1400'::bigint)
               Filter: (document OPERATOR(helio_api_catalog.@~) '{ "a.b" : "^.+$" }'::helio_core.bson)
               ->  Bitmap Index Scan on _id_
                     Index Cond: (shard_key_value = '1400'::bigint)
(10 rows)

